src/components/verification-grid/verification-grid.ts

```diff
@@ -933,6 +1054,158 @@ export class VerificationGridComponent extends WithShoelace(AbstractComponent(Li
     this.currentSubSelection = gridTiles.filter((tile) => tile.selected).map((tile) => tile.model);
   }
 
+  /**
+   * A common method that can be used to create consistent selection behavior
+   * across the different selection methods (click, alt, arrow, tab & highlight)
+   */
+  private processSelection(
+    tileIndices: number | number[],
+    { additive = false, toggle = false, range = false, focus = false }: SelectionOptions = {},
+  ): void {
+    if (!this.canSubSelect()) {
+      return;
+    }
+
+    this.isCurrentAutoSelected = false;
+
+    let selectionBehavior = this.selectionBehavior;
+    if (selectionBehavior === "default") {
+      selectionBehavior = this.isMobileDevice() ? "tablet" : "desktop";
+    }
+
+    const selectionIndex = tileIndices;
+
+    // in desktop mode, unless the ctrl key is held down, clicking an element
+    // removes all other selected items
+    // while it is not possible to press the ctrl key on a tablet, the user can
+    // still overwrite the selection behavior using the selection-behavior
+    // attribute. Therefore, we have to check that we are not on explicitly
+    // using tablet selection mode.
+    if (selectionBehavior === "desktop" && !toggle && !additive && !focus) {
+      this.removeSubSelection();
+    }
+
+    const iterableIndices = Array.isArray(selectionIndex) ? selectionIndex : [selectionIndex];
+    for (const tileIndex of iterableIndices) {
+      // If the "focus" selection behavior is set, we want to focus the tile but
+      // not select it.
+      if (focus) {
+        this.focusTile(tileIndex);
+        this.focusHead = tileIndex;
+      } else if (range) {
+        // if the user has never selected an item before, the multiSelectHead will be "null"
+        // in this case, we want to start selecting from the clicked tile
+        this.rangeSelectionHead ??= tileIndex;
+        const selectionTail = tileIndex;
+
+        this.addSubSelectionRange(this.rangeSelectionHead, selectionTail);
+      } else if (additive) {
+        this.selectTile(tileIndex);
+        this.rangeSelectionHead = tileIndex;
+      } else {
+        // if we reach this point, we know that the user is not performing a
+        // range selection because range selection performs an early return
+        this.toggleTileSelection(tileIndex);
+        this.rangeSelectionHead = tileIndex;
```

Comment:
finding this section a little hard to read, maybe some examples in comments would help?

i think range is for a shift+click, and additive is for ctrl+click.

what happens when both range and additive are true?, e.g. ctrl+shift click - or is range simply the priority? Or does range imply additive?

Reply:
Ctrl + click is "toggle" because ctrl + clicking on a currently selected tile will de-select it.

"additive" mode is used for the highlight selection box (although it probably shouldn't be, because in Window's explorer ctrl + highlight can toggle/deselect items).